drivers/microsol-apc.c

```diff
@@ -0,0 +1,184 @@
+/* microsol-apc.c - Driver for APC Back-UPS BR UPS hardware
+
+   Copyright (C) 2004  Silvino B. Magalhães    <sbm2yk@gmail.com>
+                 2019  Roberto Panerai Velloso <rvelloso@gmail.com>
+                 2021  Ygor A. S. Regados      <ygorre@tutanota.com>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+
+   2021/03/19 - Version 0.70 - Initial release, based on solis driver
+
+*/
+
+#include <ctype.h>
+#include <stdio.h>
+#include "main.h"
+#include "serial.h"
+#include "nut_float.h"
+#include "timehead.h"
+
+#include "microsol-common.h"
+#include "microsol-apc.h"
+
+#define DRIVER_NAME	"APC Back-UPS BZ series UPS driver"
+#define DRIVER_VERSION	"0.68"
```

Comment:
Please check if the DRIVER_NAME is right? It seems to mismatch the description in the file heading a bit. Also, for a new driver makes sense to start with new numbering, unless it is very closely related to this version (minus one?) of another driver used as a starting point?

Reply:
Just pushed a new commit fixing the DRIVER_NAME. As for the version, does the project have any guidelines for the numbering? If not, I would use version 1.0.

Reply:
Good question, I don't remember any such guidance. Looking at existing sources, most drivers have `0.XY` numbers with a few claiming non-zero major number and a few having one digit after decimal. I don't think there were particular reasons for any number, other than increasing the value as features/logic were changed or added. Probably the zero reflects a "young" driver that should graduate into higher numbers as many devices are proven to be supported and bugless, but I think this was never automated or pursued manually.

@clepple @aquette : any insights on DRIVER_VERSION assignments?

Reply:
Indeed X.YZ. probably around the new-driver doc, or simply inherited from what Russel did setup. Would need official formalization and possible enforcement mister new Leader ;)